handlers/mentionHandler.js

```diff
@@ -0,0 +1,30 @@
+const Discord = require('discord.js');
+const MentionsList = {
+    EVERYONE: '@everyone',
+    HERE: '@here'
+}
+const MENTION_PERMISSION = 'MENTION_EVERYONE';
+const EMBED_COLOR = '#eb1540';
+const EMBED_TITLE = 'Nie wołaj wszystkich!';
+const EMBED_THUMBNAIL = 'https://i.imgur.com/nREiJww.png';
+const EMBED_DESCRIPTION = '\nRozumiemy, że potrzebujesz pomocy, ale nie wszyscy chcą zostać o tym powiadomieni.\n Jest nas tutaj dużo i nie ma sensu, aby każdy dostał bezpośrednio taką informację.\n Nie trudno sobie wyobrazić jak irytujące byłoby, gdyby każdy wołał wszystkich do każdego tematu.\n Dlatego zadaj pytanie i po prostu poczekaj - jeśli ktoś będzie wiedział i mógł, to na pewno spróbuje odpowiedzieć.';
+
+module.exports = {
+    
+    handler: function (message){
+        const pingEmbed = new Discord.MessageEmbed()
+        .setColor(EMBED_COLOR)
+        .setTitle(EMBED_TITLE)
+        .setThumbnail(EMBED_THUMBNAIL)
+        .setDescription(EMBED_DESCRIPTION);
+        const mentions = message.content.includes(MentionsList.EVERYONE) || message.content.includes(MentionsList.HERE);
+        try {
+            const hasPermission = message.member.hasPermission(MENTION_PERMISSION);
+            if (mentions && !hasPermission) {
+                message.author.send(pingEmbed);
+            }
+        } catch(error) {
+            console.log(error);
```

Comment:
Skoro już jest tu ten `try..catch`, to dodał bym informację do logowania i użył metody `error` zamiast `log`, np. `console.error('mentionHandler(..) send error:', error)`. Ponadto, jeśli oczekiwane jest, że sypnąć może się tylko wywołanie `message.author.send`, to wyrzuciłbym z bloku `try` (przeniósł wyżej) tworzenie zmiennej `hasPermission` (tam się raczej nie ma co wysypać?). Wtedy jeśli się coś sypnie, to będzie można jednoznacznie stwierdzić, że to wysyłanie i informacja w logu na to wskaże. Deklaracja tej zmiennej `hasPermission` w bloku `try` - przy założeniu, że ona się nie sypie - IMO trochę zaciemnia przed czym chroni cały `try..catch`.

Reply:
Rzucany jest błąd `hasPermission is not defined`

Reply:
Problem w tym że discord.js niezbyt pozwala to zifować (albo poprostu jestem zbyt głupi by to zrobić :V)

Reply:
W jakim sensie nie pozwala?

Zamień:

```js
try{ //needed, if someone sends private message to bot, it crashes without try..catch
    const hasPermission = message.member.hasPermission(MENTION_PERMISSION);
```
na:
```js
const hasPermission = message.member.hasPermission && message.member.hasPermission(MENTION_PERMISSION);
```
, albo:
```js
const hasPermission = typeof message.member.hasPermission === 'function' ? 
  message.member.hasPermission(MENTION_PERMISSION) : false;
```